modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/ReplicatorGroupImpl.java

```diff
@@ -119,14 +122,16 @@ public boolean addReplicator(final PeerId peer, final ReplicatorType replicatorT
         assert client != null;
 
         if (!client.connect(peer)) {
-            if (!failureReplicators.containsKey(peer)) {
+            boolean added = failureReplicatorsSetToPreventLogFlooding.add(peer);
+
+            if (added) {
                 LOG.error("Fail to check replicator connection to peer={}, replicatorType={}.", peer, replicatorType);
             }
 
             this.failureReplicators.put(peer, replicatorType);
             return false;
         } else {
-            failureReplicators.remove(peer);
+            failureReplicatorsSetToPreventLogFlooding.remove(peer);
```

Comment:
Why have you completely removed this line `failureReplicators.remove`? You still add the replicator to `failureReplicators` in `if` block but now don't remove in `else`